source/frontend/src/utils/getKeycloakEventHandler.ts

```diff
@@ -15,24 +24,39 @@ const getKeycloakEventHandler = (keycloak: Keycloak, onRefresh: () => void) => {
     eventType: AuthClientEvent,
     error?: AuthClientError | undefined,
   ) => {
-    if (eventType === 'onAuthSuccess') {
-      store.dispatch(saveJwt(keycloak.token ?? ''));
-      // store the currently logged user so that telemetry spans can be traced back to user actions
-      const userDetails = getUserDetailsFromKeycloakToken(keycloak.tokenParsed);
-      user.getUserManager().setUser(userDetails);
-    } else if (eventType === 'onAuthRefreshSuccess') {
-      onRefresh();
-      store.dispatch(saveJwt(keycloak.token ?? ''));
-    } else if (eventType === 'onAuthLogout' || eventType === 'onTokenExpired') {
-      store.dispatch(clearJwt());
-      user.getUserManager().clearUser();
-    } else if (eventType === 'onReady') {
-      store.dispatch(setKeycloakReady(true));
-    } else {
-      toast.error(errorMessage);
-      console.debug(`keycloak event: ${eventType} error ${JSON.stringify(error)}`);
-    }
+    // Track keycloak authentication events with browser telemetry
+    runWithSpan('process keycloak.event', { component: 'keycloak' }, span => {
+      const attributes: Attributes = {
+        [ATTR_EVENT_NAME]: eventType,
+        [ATTR_CODE_FUNCTION]: 'getKeycloakEventHandler',
+        'user.is_authenticated': keycloak?.authenticated ?? false,
+      };
+
+      if (eventType === 'onAuthSuccess') {
+        store.dispatch(saveJwt(keycloak.token ?? ''));
+        // store the currently logged user so that telemetry spans can be traced back to user actions
+        const userDetails = getUserDetailsFromKeycloakToken(keycloak.tokenParsed);
+        user.getUserManager().setUser(userDetails);
+        SpanEnrichment.enrichWithKeycloakToken(attributes, keycloak);
+      } else if (eventType === 'onAuthRefreshSuccess') {
+        onRefresh();
+        store.dispatch(saveJwt(keycloak.token ?? ''));
+        SpanEnrichment.enrichWithKeycloakToken(attributes, keycloak);
+      } else if (eventType === 'onAuthLogout' || eventType === 'onTokenExpired') {
+        store.dispatch(clearJwt());
```

Comment:
Can we get something added to span in this case? These two eventTypes are likely what a user will see when they are forcibly logged out.

Reply:
We are logging the event name when all the events happen (in line 56, outside of all the if/then/elses). For this event in particular what extra would you want to log? the token expiration date? open to ideas

Reply:
I was thinking the error/error_description